### PR TITLE
[MRG] Parallelize pytests where applicable

### PR DESCRIPTION
**Pull request by asoplata**

### CODE REVIEW

**Comment by asoplata** [comment]
Based on comparison to other recent test runs, this appears to shave about 3 minutes off of running the tests on the Macos and Linux github action runners. This especially helps speed up local testing. I will test making the same changes to Windows after I try installing it on a local Windows machine.

File: .github/workflows/unix_unit_tests.yml [hunk 1]
```diff
@@ -58,11 +58,15 @@
...
-        - name: Test with pytest
+        - name: Test embarrassingly parallel tests with pytest
+          shell: bash -el {0}
+          run: |
+            python -m pytest ./hnn_core/tests/ -m "not already_parallel" -n auto --cov=hnn_core --cov-report=xml
+        - name: Test true parallel tests with pytest
           shell: bash -el {0}
           run: |
-            python -m pytest ./hnn_core/tests/  --cov=hnn_core --cov-report=xml
+            python -m pytest ./hnn_core/tests/ -m "already_parallel" --cov=hnn_core --cov-report=xml --cov-append
```
Comment (jasmainak) [inline .github/workflows/unix_unit_tests.yml:68]: maybe we should add this to the Makefile: https://github.com/jonescompneurolab/hnn-core/blob/master/Makefile

then you use the same command locally too. E.g.:

```sh
$ make already_parallel_tests
```

Comment (jasmainak) [reply .github/workflows/unix_unit_tests.yml:68]: and the `make test` has a dependency on `already_parallel_tests` and `parallel_tests` like so:

```sh
tests : already_parallel parallel_tests
```
could you come up with slightly better names so the distinction between the two categories are clear?

Comment (asoplata) [reply .github/workflows/unix_unit_tests.yml:68]: What about `@pytest.mark.uses_mpi`? That way it is unique, easy to understand (and `grep`), but also distinct from the other valid decorator `@requires_mpi4py`.

Comment (jasmainak) [reply .github/workflows/unix_unit_tests.yml]: yes, I like that!

**Comment by dylansdaniels** [comment]
Austin and I reviewed this together and it looks good and is ready to merge.

**Pull request merged**